x/ibc/02-client/types/height.go

```diff
@@ -12,6 +13,11 @@ import (
 
 var _ exported.Height = (*Height)(nil)
 
+// IsEpochFormat checks if a chainID is in the format required for parsing epochs
+// The chainID must be in the form: `{chainID}-{version}
+// 24-host may enforce stricter checks on chainID
+var IsEpochFormat = regexp.MustCompile(`^.+[^-]-{1}[1-9][0-9]*$`).MatchString
```

Comment:
I prefer the regex to remain as is. This format is only checking if an epoch number is parsable from the chainID and not enforcing any other constraints. To do this, the rest of the chainID must be seperated from the epoch number by a single dash. That's all that the regex checks. If IBC wants to enforce further constraints on the chainID as you mentioned above, it should do so in `24-host`.

I definitely want to allow `gaia-mainnet-40` to pass since `gaia-mainnet` could be a plausible "base-chainID". I even want to allow numbers to also exist like `gaia-mainnet-3-30` since this will allow further uses of the chainID by other applications. With `gaiamainnet-3-30`, 30 is parsed as the epoch number; but some other application may need to parse out 3 for its own purposes. It could also be useful for certain non-programmatic reasons, for example the chainID may want to include the start of the gaia hash inside its chainID:

`gaia-123456-4`. Here epoch is 4, but the chain may include the hash `123456` for setup purposes or something like that.

Reply:
sounds reasonable to me. Would like an ack from @cwgoes since this [comment](https://github.com/cosmos/cosmos-sdk/pull/7280#discussion_r487553886) suggests a stricter format. 

It is worthwhile noting a chainID cannot end in a dash with the current construction. Is there a reason for this?

Reply:
If a chainID ends in a dash, then yes it will just default to 0 epoch which seems reasonable to me.

Either way, agree with your later comment that further checks can be done in a followup since changing it will only require changing that regex and potentially some test chainIDs

Reply:
Hmm. `gaia-mainnet-40` does seem like a reasonable chain ID. I'm worried about examples like `gaia-epoch-2-launch-date-2020-10-05`, where we would parse `05` as the epoch, which is clearly not intended. Admittedly this example is contrived, and it's also possible for someone to use a valid chain ID even under the stricter regex - `gaia-0203` - which was not intended as an epoch.

Upon reflection banning valid chain IDs seems slightly worse than risking misparsing; all the examples I can come up with are pretty contrived and it's not dangerous anyways, just a little odd (potentially dangerous if the epoch isn't changed on an upgrade, but at that point it's the fault of the upgrade author) - so given that I think the current regex is fine.

Thanks @colin-axner for the diligence!

Reply:
> I'm worried about examples like gaia-epoch-2-launch-date-2020-10-05, where we would parse 05 as the epoch, which is clearly not intended.

Technically this wouldn't pass since I don't allow epochs to start with `0` but your point is generally taken. This is why I started out by enforcing that the chainID was formatted like so: `chainID-epoch-3`. This would ensure that the chainID was explicitly being constructed for epoch parsing. I'm happy to add that back in, in a subsequent PR if people prefer this.

Reply:
I'd be a fan of adding back the `-epoch-` except renaming to `-version-`.

Reply:
The annoying aspect of that format is that it's not compatible with the current Hub chain ID format.

Reply:
this sounds like a question for the ecosystem to answer